qiskit/synthesis/evolution/product_formula.py

```diff
@@ -58,15 +89,27 @@ def __init__(
         # user-provided atomic evolution, stored for serialization
         self._atomic_evolution = atomic_evolution
         self._cx_structure = cx_structure
+        self._wrap = wrap
 
         # if atomic evolution is not provided, set a default
         if atomic_evolution is None:
-            atomic_evolution = partial(_default_atomic_evolution, cx_structure=cx_structure)
+            self.atomic_evolution = partial(
+                _default_atomic_evolution, cx_structure=cx_structure, wrap=wrap
+            )
+
+        elif len(inspect.signature(atomic_evolution).parameters) == 2:
+
+            def wrap_atomic_evolution(output, operator, time):
+                definition = atomic_evolution(operator, time)
+                output.compose(definition, wrap=wrap, inplace=True)
```

Comment:
I was not sure whether we should also do the explicit `Gate` conversion here?
```suggestion
                definition = atomic_evolution(operator, time)
                gate = definition.to_gate()
                output.append(gate, qargs=output.qubits, copy=False)
```
I believe, this will affect the display label. Maybe one can add `label=definition.name` to the `.to_gate()` call to ensure whatever name the user might have set, is preserved? :thinking:

Reply:
After an offline discussion, we are leaving this with `compose`.